File: ConsoleApp1/ConsoleApp1/QuyenSachGiaoTrinh.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace QLTV
{
	class QuyenSachGiaoTrinh:DauSachGiaoTrinh,DacTrungQuyenSach
	{
		private int sMaQuyenSach;
		private bool bDaMuonHayChua;
		private Date dtNgayMuon;
		private Date dtNgayPhaiTra;
		private int iTinhTrang;          //0 là cũ , 1 là mới
		public int MaQuyenSach
		{
			get { return this.sMaQuyenSach; }
		}
		public bool DaMuonHayChua
		{
			get { return this.bDaMuonHayChua; }
			set { this.bDaMuonHayChua = value; }
		}
		public Date NgayMuon
		{
			get { return this.dtNgayMuon; }
			set { this.dtNgayMuon = value; }
		}
		public Date NgayPhaiTra
		{
			get { return this.dtNgayPhaiTra; }
			set { this.dtNgayPhaiTra = value; }
		}
		public QuyenSachGiaoTrinh() : base()
		{
			this.bDaMuonHayChua = true;
			this.dtNgayMuon = new Date();
			this.dtNgayPhaiTra = new Date();
		}
		public int TinhTrang
		{
			get { return this.iTinhTrang; }
			set { this.iTinhTrang = value; }
		}
		public QuyenSachGiaoTrinh(string MaDauSach, string Tua, string NXB, string TacGia, double Gia, string BoMon, int TinhTrang,int MaQuyenSach) : base(MaDauSach, Tua, NXB, TacGia, Gia,BoMon)
		{
			this.sMaQuyenSach = MaQuyenSach;
			this.iTinhTrang = TinhTrang;	
			this.bDaMuonHayChua = true;
			this.dtNgayMuon = new Date();
			this.dtNgayPhaiTra = new Date();
			this.XacDinhThue();
			this.TinhThanhTien();
		}
		public override void Nhap()
		{
			base.Nhap();
			Console.Write("Tinh trang ( 0 la cu || 1 la moi ): ");
			this.iTinhTrang = Convert.ToInt32(Console.ReadLine());
		}
		public override void Xuat()
		{
			base.Xuat();
			if (this.iTinhTrang == 0)
			{
				Console.WriteLine(" Tinh Trang: Cu ");
			}
			else
			{
				Console.WriteLine(" Trinh trang: Moi ");
			}
			if (this.bDaMuonHayChua == false)
			{
				Console.WriteLine(" Da duoc muon! ");
				Console.WriteLine("Ngay het duoc muon : " + this.dtNgayMuon.Ngay + "/" + this.dtNgayMuon.Thang + "/" + this.dtNgayMuon.Nam);
				Console.WriteLine("Ngay het phai tra : " + this.dtNgayPhaiTra.Ngay + "/" + this.dtNgayPhaiTra.Thang + "/" + this.dtNgayPhaiTra.Nam);
			}
			else
			{
				Console.WriteLine(" Nam trong thu vien!  ");
			}
			Console.WriteLine(" Ma quyen sach: " + this.sMaQuyenSach);
			
		}
		public override void TinhThanhTien()
		{
			if (this.iTinhTrang == 0)
			{
				this.dThanhTien = (this.dGia / 2) + (this.dThue*this.dGia/100);
			}
			else
			{
				this.dThanhTien = this.dGia + (this.dThue*this.dGia/100);
			}
		}
		public void MuonNgayHomNay(Date Today)
		{
			this.bDaMuonHayChua = false;
			this.dtNgayMuon = Today;
			this.dtNgayPhaiTra = (Today + 120);
		}
		public void TraSach()
		{
			this.bDaMuonHayChua = true;
			this.dtNgayMuon = new Date();
			this.dtNgayPhaiTra = new Date();
		}
		public void XDNgayTra()
		{
			this.dtNgayPhaiTra = this.dtNgayMuon + 120;
		}
	}
}

File: Quan_Ly_Thu_Vien/src/QuyenSachGiaoTrinh.java
import java.util.Scanner;

public class QuyenSachGiaoTrinh extends DauSachGiaoTrinh implements DacTrungQuyenSach
{
	private int sMaQuyenSach;
	private boolean bDaMuonHayChua;
	private Date dtNgayMuon;
	private Date dtNgayPhaiTra;
	private int iTinhTrang;          
	public int getsMaQuyenSach() {
		return sMaQuyenSach;
	}
	public void setsMaQuyenSach(int sMaQuyenSach) {
		this.sMaQuyenSach = sMaQuyenSach;
	}
	public boolean getbDaMuonHayChua() {
		return bDaMuonHayChua;
	}
	public void setbDaMuonHayChua(boolean bDaMuonHayChua) {
		this.bDaMuonHayChua = bDaMuonHayChua;
	}
	public Date getdtNgayMuon() {
		return dtNgayMuon;
	}
	public void setdtNgayMuon(Date dtNgayMuon) {
		this.dtNgayMuon = dtNgayMuon;
	}
	public Date getdtNgayPhaiTra() {
		return dtNgayPhaiTra;
	}
	public void setdtNgayPhaiTra(Date dtNgayPhaiTra) {
		this.dtNgayPhaiTra = dtNgayPhaiTra;
	}
	public int getiTinhTrang() {
		return iTinhTrang;
	}
	public void setiTinhTrang(int iTinhTrang) {
		this.iTinhTrang = iTinhTrang;
	}
	public QuyenSachGiaoTrinh()
	{
		super();
		this.bDaMuonHayChua = true;
		this.dtNgayMuon = new Date();
		this.dtNgayPhaiTra = new Date();
	}
	public QuyenSachGiaoTrinh(String MaDauSach, String Tua, String NXB, String TacGia, double Gia, String BoMon, int TinhTrang,int MaQuyenSach) 
	{
		super(MaDauSach, Tua, NXB, TacGia, Gia,BoMon);
		this.sMaQuyenSach = MaQuyenSach;
		this.iTinhTrang = TinhTrang;	
		this.bDaMuonHayChua = true;
		this.dtNgayMuon = new Date();
		this.dtNgayPhaiTra = new Date();
		this.XacDinhThue();
		this.TinhThanhTien();
	}
	@SuppressWarnings("resource")
	@Override
	public void Nhap()
	{
		super.Nhap();
		Scanner sc = new Scanner(System.in);
		System.out.print("Tinh trang ( 0 la cu || 1 la moi ): ");
		this.iTinhTrang = sc.nextInt();
	}
	@Override
	public void Xuat()
	{
		super.Xuat();
		if (this.iTinhTrang == 0)
		{
			System.out.println(" Tinh Trang: Cu ");
		}
		else
		{
			System.out.println(" Trinh trang: Moi ");
		}
		if (this.bDaMuonHayChua == false)
		{
			System.out.println(" Da duoc muon! ");
			System.out.println(" Ngay muon : " + this.dtNgayMuon.getiNgay() + "/" + this.dtNgayMuon.getiThang() + "/" + this.dtNgayMuon.getiNam());
			System.out.println(" Ngay phai tra : " + this.dtNgayPhaiTra.getiNgay() + "/" + this.dtNgayPhaiTra.getiThang() + "/" + this.dtNgayPhaiTra.getiNam());
		}
		else
		{
			System.out.println(" Nam trong thu vien!  ");
		}
		System.out.println(" Ma quyen sach: " + this.sMaQuyenSach);
//		System.out.println(" Thanh tien: "+this.dThanhTien);
	}
	@Override
	public void TinhThanhTien()
	{
		if (this.iTinhTrang == 0)
		{
			this.dThanhTien = (this.dGia / 2) + (this.dThue*this.dGia/100);
		}
		else
		{
			this.dThanhTien = this.dGia + (this.dThue*this.dGia/100);
		}
	}
	public void MuonNgayHomNay(Date Today)
	{
		this.bDaMuonHayChua=false;
		this.dtNgayMuon = Today;
		this.dtNgayPhaiTra = Date.PlusDay(Today,120);
	}
	public void TraSach()
	{
		this.bDaMuonHayChua = true;
		this.dtNgayMuon = new Date();
		this.dtNgayPhaiTra = new Date();
	}
	public void XDNgayTra()
	{
		this.dtNgayPhaiTra = Date.PlusDay(this.dtNgayMuon,120);
	}
}
